Update deploying.md (#1356)
Techtonica/curriculum: deploying/deploying.md

- [Video - What is Heroku (4 mins watch)](https://youtu.be/r5ZUQvl9BtE)
- [Techtonica Slides: Deploying](https://docs.google.com/presentation/d/1Enwhd9hl1fn1-afMXJ6xvkJm5SDJpHjfQoA7s2znHpw/edit?usp=sharing)

> - [Deploy Node.JS Apps to Heroku (10 mins Video)](https://youtu.be/AZNFox2CvBk)

Reviewer: I think it is better to add one more material on `how to run front-end and back-end on a same server`.

Author: I found this one: [https://www.twilio.com/blog/react-app-with-node-js-server-proxy]
Do tell if you want me to include this or you have a better resource

Reviewer 2: I don't understand why - this outline is specifically about heroku.

Author: Ok, I won't add it.

- [Video - What is Heroku (4 mins watch)](https://youtu.be/r5ZUQvl9BtE)
- [Techtonica Slides: Deploying](https://docs.google.com/presentation/d/1Enwhd9hl1fn1-afMXJ6xvkJm5SDJpHjfQoA7s2znHpw/edit?usp=sharing)
- [Deploy Node.JS Apps to Heroku (10 mins Video)](https://youtu.be/AZNFox2CvBk)

> - []

Reviewer 3: Why you added this ?

Now we'll work on deploying your app to Heroku.

We'll be combining your front-end (create-react-app) with your back-end (express) and deploying it to Heroku. Your front-end contains "static" JavaScript files -- when you deploy to Heroku, it turns your whole React app into a couple of static files that it will serve to the browser. No matter what data you have in the database, these files will always be the same.

Your backend, on the other hand, is dynamic -- when you make an API request, the backend runs javascript code to do things like reading and writing to a database. Unlike the React app, which always serves the same files to the browser, the backend will serve different information to the browser depending on what's in the database. We're going to combine your dynamic code (express), with your static code (create-react-app).

> 1. cd into the React app and create 2 new directories `client` and `server` you created and move _everything_ into the new directory `client`:

Reviewer 2: Thank you for adding that, but could you rephrase the sentence now?  It doesn't make sense anymore.

Now we'll work on deploying your app to Heroku.

> We'll be combining your frontend (create-react-app) with your backend (express) and deploying it to Heroku. Your frontend contains "static" JavaScript files -- when you deploy to Heroku, it turns your whole React app into a couple of static files that it will serve to the browser. No matter what data you have in the database, these files will always be the same.

Reviewer 3: ```suggestion
We'll be combining your front-end(create-react-app) with your back-end (express) and deploying it to Heroku. Your front-end contains "static" JavaScript files -- when you deploy to Heroku, it turns your whole React app into a couple of static files that it will serve to the browser. No matter what data you have in the database, these files will always be the same.
```

> Your backend, on the other hand, is dynamic -- when you make an API request, the backend runs javascript code to do things like reading and writing to a database. Unlike the React app, which always serves the same files to the browser, the backend will serve different information to the browser depending on what's in the database. We're going to combine your dynamic code (express), with your static code (create-react-app).

Reviewer: After this, the first point should be **cd into your app and create 2 new directories `client` and `server`**. The reason for this is we don't now what kind of directory structure they have followed ( like React-app inside the Back-end or vice-versa). It will be more better if you add a image of the final directory structure after 2nd point.

Reviewer 2: Hi @Aayush-1999 , it isn't part of the issue to fix this, but thanks for pointing it out.  I will make sure it gets made into a new issue.

Your backend, on the other hand, is dynamic -- when you make an API request, the backend runs javascript code to do things like reading and writing to a database. Unlike the React app, which always serves the same files to the browser, the backend will serve different information to the browser depending on what's in the database. We're going to combine your dynamic code (express), with your static code (create-react-app).

> 1. cd into the React app and create 2 new directories `client` and `server` you created and move _everything_ into the  new directory `client`:

Reviewer 3: ```suggestion
1. cd into the React app and create 2 new directories `client` and `server` you created and move _everything_ into the new directory `client`:
```

```json
    "start": "node server/server.js",

>     "heroku-postbuild": "npm install --only=dev --no-shrinkwrap && npm run build"

Reviewer 2: 💜

    Through the Heroku web UI, create a new Application.
    Once you create the app, add the Postgres add-on by going to the Resources tab
    and searching in the "Add-ons" search box for Postgres.
    Click the "Heroku Postgres" option. Finally, select the free version and click "Provision". 

>     Or you can create the new Application fom the CLI. To create an application with name,install the Heroku CLI and run the following command

Reviewer 3: ```suggestion
    Or you can create the new Application from the CLI. To create an application with name, install the Heroku CLI and run the following command
```

>     Or you can create the new Application from the CLI. To create an application with name, install the Heroku CLI and run the following command

Reviewer 3: ```suggestion
    Or you can create a new Application from the CLI. To create an application with name, install the Heroku CLI and run the following command
```

13. Deploy your app!
    `git push heroku master`
    This takes a long time.

>     This will print the URL your app was deployed to. Try going to it! If something goes run, use `heroku logs --tail` to debug.

Reviewer 3: ```suggestion
    This will print the URL your app was deployed to. Try going to it! If something goes wrong, use `heroku logs --tail` to debug.
```

- [Tutorial - Heroku Dev Center Deployment](https://devcenter.heroku.com/articles/deploying-nodejs)
- [Overview of Deployment Options - MDN Express & Node Deployment](https://developer.mozilla.org/en-US/docs/Learn/Server-side/Express_Nodejs/deployment)

> - [Code School Tutorial(requires sign-in) - Domain Names, Name Servers and Cloud Based Servers](https://www.pluralsight.com/paths/html5)

Reviewer 3: You have only changed the link but forgot to change the Label.

Author: Can you suggest a proper label name for it?

Reviewer 3: `HTML5`. And change the description too.

Reviewer 2: Please only provide free courses.  I think this one requires a membership, plus I don't think HTML5 it is a good replacement for  a tutorial about domain names and servers.

Author: Reversed the changes again

Author: @alodahl The link was of HTML5 only!

Reviewer 2: Yes, it is wrong and outdated.  Please replace it with a comparable tutorial or delete the whole line.

> - [Code School Tutorial(requires sign-in) - Domain Names,Name Servers and Cloud Based Servers](https://www.codeschool.com/beginners-guide-to-web-development/deploying-your-first-website)

Reviewer 3: ```suggestion
- [Code School Tutorial(requires sign-in) - Domain Names, Name Servers and Cloud Based Servers](https://www.codeschool.com/beginners-guide-to-web-development/deploying-your-first-website)
```

> - [Code School Tutorial(requires sign-in) - Domain Names,Name Servers and Cloud Based Servers](https://www.codeschool.com/beginners-guide-to-web-development/deploying-your-first-website)

Reviewer 3: This link is redirecting to `https://www.pluralsight.com/paths/html5` so better to switch it with that or find other one.

- [Tutorial - Heroku Dev Center Deployment](https://devcenter.heroku.com/articles/deploying-nodejs)
- [Overview of Deployment Options - MDN Express & Node Deployment](https://developer.mozilla.org/en-US/docs/Learn/Server-side/Express_Nodejs/deployment)

> - [Code School Tutorial(requires sign-in) - Domain Names,Name Servers, and Cloud Based Servers](https://www.codeschool.com/beginners-guide-to-web-development/deploying-your-first-website)

Reviewer 2: 👍🏼
